modules/basics/src/main/java/com/opengamma/strata/basics/currency/FxRatesArray.java

```diff
@@ -0,0 +1,473 @@
+
+  /**
+   * Returns an array of FX rates for a currency pair.
+   * <p>
+   * The rates are the rates from the base currency to the counter currency
+   * as defined by this formula: {@code (1 * baseCurrency = fxRate * counterCurrency)}.
+   *
+   * @param currencyPair  the currency pair
+   * @param rates  the FX rates for the currency pair
+   * @return an array of FX rates for a currency pair
+   */
+  public static FxRatesArray of(CurrencyPair currencyPair, DoubleArray rates) {
+    return new FxRatesArray(currencyPair, rates);
+  }
+
+  /**
+   * Returns an array of FX rates for a currency pair.
+   * <p>
+   * The rates are the rates from the base currency to the counter currency
+   * as defined by this formula: {@code (1 * baseCurrency = fxRate * counterCurrency)}.
+   *
+   * @param base  the base currency of the pair
+   * @param counter  the counter currency of the pair
+   * @param rates  the FX rates for the currency pair
+   * @return an array of FX rates for a currency pair
+   */
+  public static FxRatesArray of(Currency base, Currency counter, DoubleArray rates) {
+    return new FxRatesArray(CurrencyPair.of(base, counter), rates);
+  }
+
+  /**
+   * Returns the FX rate for a scenario.
+   *
+   * @param scenarioIndex  the index of the scenario
+   * @return the FX rate for the specified scenario
+   */
+  @Override
+  public FxRate getValue(int scenarioIndex) {
+    return FxRate.of(pair, rates.get(scenarioIndex));
+  }
+
+  @Override
+  public int getScenarioCount() {
+    return rates.size();
+  }
+
+  /**
+   * Returns the FX rate for the specified currency pair and scenario index.
+   * <p>
+   * The rate returned is the rate from the base currency to the counter currency
+   * as defined by this formula: {@code (1 * baseCurrency = fxRate * counterCurrency)}.
+   * <p>
+   * This will return the rate or inverse rate, or 1 if the two input currencies are the same.
+   * <p>
+   * This method is more efficient than {@link #getValue(int)} as it doesn't create an instance
+   * of {@link FxRate} for every invocation.
+   *
+   * @param baseCurrency  the base currency, to convert from
+   * @param counterCurrency  the counter currency, to convert to
+   * @param scenarioIndex  the index of the scenario for which rates are required
+   * @return the FX rate for the currency pair
+   * @throws IllegalArgumentException if no FX rate could be found
+   */
+  public double fxRate(Currency baseCurrency, Currency counterCurrency, int scenarioIndex) {
+    if (baseCurrency.equals(counterCurrency)) {
+      return 1d;
+    }
+    if (baseCurrency.equals(pair.getBase()) && counterCurrency.equals(pair.getCounter())) {
+      return rates.get(scenarioIndex);
+    }
+    if (counterCurrency.equals(pair.getBase()) && baseCurrency.equals(pair.getCounter())) {
+      return 1d / rates.get(scenarioIndex);
+    }
+    throw new IllegalArgumentException("Unknown rate: " + baseCurrency + "/" + counterCurrency);
+  }
```

Comment:
I imagine there will be bulk conversion methods eventually, but best to add them as needed.